stargz/stargz_test.go

```diff
@@ -471,10 +504,18 @@ func dir(d string) tarEntry {
 	})
 }
 
-func file(name, contents string, extraAttr ...interface{}) tarEntry {
+// xAttr are extended attributes to set on test files created with the file func.
+type xAttr map[string]string
+
+func file(name, contents string, opts ...interface{}) tarEntry {
 	return tarEntryFunc(func(tw *tar.Writer) error {
-		if len(extraAttr) > 0 {
-			return errors.New("unsupported extraAttr")
+		var xattrs xAttr
+		for _, opt := range opts {
+			if v, ok := opt.(xAttr); ok {
+				xattrs = v
+			} else {
+				return errors.New("unsupported opt")
```

Comment:
return fmt.Errorf("unknown opt type %T", opt)

But more common is a type switch:

```go
    switch opt := opt.(type) {
    case xAttr:
         xAttrs = opt
     default:
         return fmt.Errorf(....)
```